### DISCUSSION TITLE
How can I make the most recently published item of a content type be the home page?

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#11461] [Q&A] [answered]
I have a content type called BoardPacket and a new BoardPacket item is created each month. I want the home page of the site to always show the most recently published BoardPacket item. What would be a good way to do that? I thought maybe the HomeRoute module (https://docs.orchardcore.net/en/latest/docs/reference/modules/HomeRoute/) could be used to do this but I have this feature enabled and can't find any settings for it.

I thought about creating a widget that used `await Orchard.GetRecentContentItemsByContentTypeAsync("BoardPacket", 1);` to get the most recent item and then `@await DisplayAsync(itemContent)` to display the item but I'm wondering if there is a better way to do this.

**Comment by ns8482e** [comment, marked as answer]
HomeRoute allows you to set a ContentItem as Home page of the site, But I guess you still need to edit the content item manually to set it as home every month. 

Your best option is to create HomeController with route "~/"  and query the content that you want to display on home page

**Reply by DrewBrasher** [reply]
Thank you. I've got it mostly working like this except that it is only rendering the main content item and not the ListPart. Here is my controller:
```
public class HomeController : Controller
{
    private IOrchardHelper _orchard;
    public HomeController(IOrchardHelper orchard)
    {
        _orchard = orchard;
    }

    [Route("~/")]
    public async Task<IActionResult> Index(string id)
    {
        var viewModel = (await _orchard.GetRecentContentItemsByContentTypeAsync("BoardPacket", 1)).FirstOrDefault();

        return View(viewModel);
    }
}
```
and here is my view:
```
@model OrchardCore.ContentManagement.ContentItem
@await Orchard.DisplayAsync(Model)
```

**Reply by DrewBrasher** [reply]
I looked at the ItemController in the OrchardCore.Contents module and changed mine to use the display manager like that and it is working now. In case anyone else comes looking for this here is my working controller:
```
using Microsoft.AspNetCore.Mvc;
using OrchardCore;
using OrchardCore.ContentManagement.Display;
using OrchardCore.DisplayManagement.ModelBinding;
using System.Linq;
using System.Threading.Tasks;

namespace MyTheme.Controllers;

public class HomeController : Controller
{
    private IOrchardHelper _orchard;
    private readonly IContentItemDisplayManager _contentItemDisplayManager;
    private readonly IUpdateModelAccessor _updateModelAccessor;
    public HomeController(IOrchardHelper orchard,
            IContentItemDisplayManager contentItemDisplayManager,
            IUpdateModelAccessor updateModelAccessor)
    {
        _orchard = orchard;
        _updateModelAccessor = updateModelAccessor;
        _contentItemDisplayManager = contentItemDisplayManager;
    }

    [Route("~/")]
    public async Task<IActionResult> Index(string id)
    {
        var mostRecentItem = (await _orchard.GetRecentContentItemsByContentTypeAsync("BoardPacket", 1)).FirstOrDefault();
        var viewModel = await _contentItemDisplayManager.BuildDisplayAsync(mostRecentItem, _updateModelAccessor.ModelUpdater);

        return View(viewModel);
    }
}
```
And all I have in the Index.cshtml view is:
```
@await DisplayAsync(Model)
```
